src/Omnipay/WorldpayCGHosted/Message/PurchaseRequest.php

```diff
@@ -203,6 +206,97 @@ public function setPaymentType($value)
         return $this->setParameter('paymentType', $value);
     }
 
+    /**
+     * @return \DateTimeInterface
+     */
+    public function getTokenExpiry()
+    {
+        return $this->getParameter('tokenExpiryDate');
+    }
+
+    /**
+     * @param \DateTimeInterface $dateTime
+     * @return AbstractRequest
+     */
+    public function setTokenExpiry(\DateTimeInterface $dateTime)
+    {
+        return $this->setParameter('tokenExpiryDate', $dateTime);
+    }
+
+    /**
+     * @return string
+     */
+    public function getCreateToken()
+    {
+        return $this->getParameter('createToken');
+    }
+
+    /**
+     * @param string $tokenType
+     * @return AbstractRequest
+     */
+    public function setCreateToken($tokenType)
+    {
+        $allowedValues = [self::TOKEN_TYPE_SHOPPER, self::TOKEN_TYPE_MERCHANT];
+        if (!in_array($tokenType, $allowedValues)) {
+            throw new \InvalidArgumentException(
+                sprintf('Allowed values for createToken are %s', implode(', ', $allowedValues))
+            );
+        }
+        return $this->setParameter('createToken', $tokenType);
+    }
+
+    /**
+     * @return string
+     */
+    public function getTokenEventReference()
+    {
+        return $this->getParameter('tokenEventReference');
+    }
+
+    /**
+     * @param string $reference
+     * @return AbstractRequest
+     */
+    public function setTokenEventReference($reference)
+    {
+        return $this->setParameter('tokenEventReference', $reference);
+    }
+
+    /**
+     * @return string
+     */
+    public function getTokenReason()
+    {
+        return $this->getParameter('tokenReason');
+    }
+
+    /**
+     * @param string $reason
+     * @return AbstractRequest
+     */
+    public function setTokenReason($reason)
+    {
+        return $this->setParameter('tokenReason', $reason);
+    }
+
+    /**
+     * @return string
+     */
+    public function getAuthenticatedShopperID()
+    {
+        return $this->getParameter('authenticatedShopperID');
+    }
+
+    /**
+     * @param string $authenticatedShopperID
+     * @return AbstractRequest
+     */
+    public function setAuthenticatedShopperID($authenticatedShopperID)
```

Comment:
The WorldPay docs define this field is the following

> a reference that uniquely identifies each shopper so that you can associate their payment tokens with them. The maximum length is 64 characters, it must contain only the ISO-latin1 subset of the UTF-8 characters, there must be no white space and it cannot start with an underscore (_). Up to 16 shopper tokens can be created against one authenticatedShopperID

Again I decided to not add any checking with this fitting with the current spec of this codebase